src/Parser.js

```diff
@@ -908,6 +908,24 @@ class Parser {
             return new ParseFuncOrArgument(
                 nucleus.text,
                 false, nucleus);
+        } else if (/^\\verb[^a-zA-Z]/.test(nucleus.text)) {
+            this.consume();
+            let arg = nucleus.text.slice(5);
+            const star = (arg.charAt(0) === "*");
+            if (star) {
+                arg = arg.slice(1);
+            }
+            // Lexer's tokenRegex is constructed to always have matching
+            // first/last characters.
+            if (arg.length < 2 || arg.charAt(0) !== arg.slice(-1)) {
+                throw new ParseError("\\verb failure to parse");
```

Comment:
This is actually more of an assertion: the regex *should* guarantee that that error is *never* thrown.   The actual error that will get thrown is `\\verb ended by end of line instead of matching delimiter` which is more descriptive.  (It doesn't mention the delimiter character, but that is not trivial to add... Let me know if I should work on it.)

Reply:
I thought the first and last characters of the `arg` were the delimiter so it seemed like a trivial change to add those characters to the error message.  If this case should never happen it's probably more important to say in the error that this is a bug and that it should be reported as such.

Reply:
Rewrote the error message.